R/binary_classification.R

```diff
@@ -138,3 +138,210 @@ fbeta_score <- function(actual, predicted, beta = 1) {
     rec <- recall(actual, predicted)
     return((1 + beta^2) * prec * rec / ((beta^2 * prec) + rec))
 }
+
+#' Binary confusion matrix
+#'
+#' \code{cmat} Calculates a binary classification confusion matrix,
+#' comparing the predicted with the actual values for the classes.
+#' Assumes that 1 is used for the positive class and 0 for the
+#' negative class.
+#' Returns a \code{data.frame} with columns corresponding to the
+#' number of True Positives (\code{tp}), False Positives (\code{fp}),
+#' True Negatives (\code{tn}), and False Negatives (\code{fn})
+#'
+#' @inheritParams params_binary
+#' @seealso \code{\link{sensitivity}} \code{\link{specificity}}
+#' @examples
+#' actual <- c(1, 1, 1, 0, 0, 0)
+#' predicted <- c(1, 0, 1, 1, 1, 1)
+#' cmat(actual, predicted)
+
+cmat <- function(actual, predicted) {
+    tbl <- as.data.frame(table(actual, predicted))
+    cm <- data.frame(tp = NA, fn = NA, fp = NA, tn = NA)
+    cm[, c("tn", "fp", "fn", "tp")] <- tbl$Freq
+    cm
+}
+
+#' Sensitivity
```

Comment:
Looking at [this page](https://en.wikipedia.org/wiki/Precision_and_recall#Definition_(classification_context)) on wikipedia, `sensitivity` is the same thing as `true_positive_rate` and `recall`. The `Metrics` package already has a function for `recall`. Could you change this function to call that function, so that the logic is only implemented in one place?

Reply:
If you fix the confusion_matrix function, you can change the internals of `recall` to use that function. Does that make sense?

Reply:
changed sensitivity to point to recall, which now uses confusion_matrix